apps/web/src/app/app/[workspaceSlug]/(dashboard)/monitors/(overview)/page.tsx

```diff
@@ -42,11 +42,13 @@ export default async function MonitorPage({
   searchParams: { [key: string]: string | string[] | undefined };
 }) {
   const search = searchParamsSchema.safeParse(searchParams);
-  const monitors = await api.monitor.getMonitorsByWorkspace.query();
-  const isLimitReached = await api.monitor.isMonitorLimitReached.query();
-
   if (!search.success) return notFound();
 
+  const [monitors, isLimitReached] = await Promise.all([
+    api.monitor.getMonitorsByWorkspace.query(),
+    api.monitor.isMonitorLimitReached.query(),
```

Comment:
This one isMonitorLImitReached could be group with the other promise.All if we don't have a monitor  we don't need to check for the limit because we are sure we haven't reached it ;)